Panic after upgrade

## What's broken?

I have just change my Mac so I reinstall wtf (with brew) at its last version but I don't know the previous installed version, I could tell  this evening.

When I run the wtf with the config located in the default place, I have this error : 

```
panic: runtime error: slice bounds out of range [-1:]

goroutine 1 [running]:
github.com/wtfutil/wtf/utils.CalculateDimensions(0xc000596520, 0xc0005963c0, 0x8, 0xc000673268)
	/private/tmp/wtfutil-20191012-32990-9vzwg7/utils/utils.go:169 +0x3f9
github.com/wtfutil/wtf/modules/cmdrunner.NewSettingsFromYAML(0xc0005841c4, 0x9, 0xc000596520, 0xc0005963c0, 0x0)
	/private/tmp/wtfutil-20191012-32990-9vzwg7/modules/cmdrunner/settings.go:40 +0x2d1
github.com/wtfutil/wtf/app.MakeWidget(0xc000538280, 0xc0000bd2f0, 0xc0005841c4, 0x9, 0xc0005963c0, 0x0, 0x26762f32837c)
	/private/tmp/wtfutil-20191012-32990-9vzwg7/app/widget_maker.go:114 +0x1f42
github.com/wtfutil/wtf/app.MakeWidgets(0xc000538280, 0xc0000bd2f0, 0xc0005963c0, 0xc000673d20, 0x1, 0x1)
	/private/tmp/wtfutil-20191012-32990-9vzwg7/app/widget_maker.go:266 +0x12a
github.com/wtfutil/wtf/app.NewWtfApp(0xc000538280, 0xc0005963c0, 0xc0003ff050, 0x24, 0x1)
	/private/tmp/wtfutil-20191012-32990-9vzwg7/app/wtf_app.go:47 +0x2e7
main.main()
	/private/tmp/wtfutil-20191012-32990-9vzwg7/main.go:69 +0x278
```

Here is my config : 

```
wtf:
  # Will fail if `colors` not present:
  colors:
  refreshInterval: 5
  mods:
    cmdrunner:
      enabled: true
      type: cmdrunner
      refreshInterval: 10
      title: 'Dev Status'
      cmd: "node"
      args: ["run.js"]
      position:
        top: 0
        left: 0
        height: 1
        width: 1
```

I tried with this example config : 

```
wtf:
 # Will fail if `colors` not present:
 colors:
 refreshInterval: 5
 mods:
   feedreader:
     enabled: true
     feeds:
     - https://news.ycombinator.com/rss
     feedLimit: 10
     focusable: false
     position:
       top: 2
       left: 2
       width: 1
       height: 1
     title: "Hacker News"
     updateInterval: 14400
``` 

and it worked.
I have no idea what I am doing wrong.

```
$ wtfutil --version
dev 
```

--- a/_sample_configs/dynamic_sizing.yml
+++ b/_sample_configs/dynamic_sizing.yml
@@ -1,4 +1,5 @@
 wtf:
+  grid:
   mods:
     battery:
       type: power

--- a/modules/cmdrunner/settings.go
+++ b/modules/cmdrunner/settings.go
@@ -27,7 +27,6 @@ type Settings struct {
 
 // NewSettingsFromYAML loads the cmdrunner portion of the WTF config
 func NewSettingsFromYAML(name string, moduleConfig *config.Config, globalConfig *config.Config) *Settings {
-
 	settings := Settings{
 		common: cfg.NewCommonSettingsFromModule(name, defaultTitle, defaultFocusable, moduleConfig, globalConfig),
 
@@ -37,7 +36,11 @@ func NewSettingsFromYAML(name string, moduleConfig *config.Config, globalConfig 
 		maxLines: moduleConfig.UInt("maxLines", 256),
 	}
 
-	settings.width, settings.height = utils.CalculateDimensions(moduleConfig, globalConfig)
+	width, height, err := utils.CalculateDimensions(moduleConfig, globalConfig)
+	if err == nil {
+		settings.width = width
+		settings.height = height
+	}
 
 	return &settings
 }

--- a/utils/utils.go
+++ b/utils/utils.go
@@ -128,15 +128,20 @@ func ParseJSON(obj interface{}, text io.Reader) error {
 }
 
 // CalculateDimensions reads the module dimensions from the module and global config. The border is already substracted.
-func CalculateDimensions(moduleConfig, globalConfig *config.Config) (int, int) {
+func CalculateDimensions(moduleConfig, globalConfig *config.Config) (int, int, error) {
+	grid, err := globalConfig.Get("wtf.grid")
+	if err != nil {
+		return 0, 0, err
+	}
+
+	cols := ToInts(grid.UList("wtf.grid.columns"))
+	rows := ToInts(grid.UList("wtf.grid.rows"))
+
 	// Read the source data from the config
 	left := moduleConfig.UInt("position.left", 0)
 	top := moduleConfig.UInt("position.top", 0)
 	width := moduleConfig.UInt("position.width", 0)
 	height := moduleConfig.UInt("position.height", 0)
-
-	cols := ToInts(globalConfig.UList("wtf.grid.columns"))
-	rows := ToInts(globalConfig.UList("wtf.grid.rows"))
 
 	// Make sure the values are in bounds
 	left = Clamp(left, 0, len(cols)-1)
@@ -157,7 +162,7 @@ func CalculateDimensions(moduleConfig, globalConfig *config.Config) (int, int) {
 	w = MaxInt(w, 0)
 	h = MaxInt(h, 0)
 
-	return w, h
+	return w, h, nil
 }
 
 // MaxInt returns the larger of x or y

